knimin/lib/data_access.py

```diff
@@ -2399,6 +2399,134 @@ def getProjectNames(self):
         sql = """SELECT project FROM project"""
         return [x[0] for x in self._con.execute_fetchall(sql)]
 
+    def get_plate_info(self, plate_id):
+        """Gets attributes of a plate by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Plate id
+
+        Returns
+        -------
+        dict
+            Plate attributes
+        """
+        sql = """SELECT * FROM pm.plate WHERE plate_id = %s"""
+        return self._con.execute_fetchdict(sql, [plate_id])[0]
+
+    def set_plate_info(self, plate_id, plate_info):
+        """Sets attributes of a plate by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Existing plate id (>0) or create new plate (0)
+        plate_info : dict
+            Plate attributes
+
+        Returns
+        -------
+        int
+            Plate id of the created / modified plate
+        """
+        keys = ', '.join([str(x) for x in plate_info.keys()])
+        values = ', '.join(['\'' + str(x) + '\'' for x in plate_info.values()])
+        sql = ''
+        if plate_id == 0:  # Create new plate
+            sql = 'INSERT INTO pm.plate (' + keys + ') VALUES (' + values + ')'
+        else:  # Modify existing plate
+            sql = 'UPDATE pm.plate SET (' + keys + ') = (' + values + ')'
+            sql += ' WHERE plate_id = ' + str(plate_id)
+        sql += ' RETURNING plate_id'
+        return self._con.execute_fetchone(sql)[0]
+
+    def get_plate_map(self, plate_id):
+        """Gets plate map by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Plate id
+
+        Returns
+        -------
+        list of tuple of (int, int, str)
+            Column id, row id, sample id
+        """
+        sql = """SELECT col, row, sample_id FROM pm.plate_sample
+                 WHERE plate_id = %s"""
+        return self._con.execute_fetchall(sql, [plate_id])
+
+    def set_plate_map(self, plate_id, plate_map):
```

Comment:
This functions is doing multiple different operations, that is confusing, and given the amount of SQL queries that can happen (96 well plate * 2 queries per well = 192 SQL queries) can make this function really slow.

If the idea of set_plate_map is to change all the contents of the plate, I think it is easier to remove the old data and insert the new data. This will be only 2 queries. However, in order to do that, we need transactions, that are currently not supported by the SQL connection handler in this repo.

@wasade any thoughts?